### ISSUE TITLE
Support for http / https URLs in LOAD DATA

### ISSUE THREAD

**Issue by okennedy** [UBOdin/mimir#340]
Spark doesn't support `http` and `https` as valid filesystems to read from (i.e., you can't load a CSV directly from http).  Therefore, something like 
```
LOAD 'https://raw.githubusercontent.com/UBOdin/mimir/master/test/data/DetectSeriesTest1.csv';
```
will fail, since Spark doesn't know what to do with an https url (this may be a protective measure to avoid spamming external servers with requests).  However, staging works fine:
```
LOAD 'https://raw.githubusercontent.com/UBOdin/mimir/master/test/data/DetectSeriesTest1.csv' WITH STAGING;
```
since the URL is downloaded into a local/S3 filesystem before anything happens.

-----

In summary, we should 
- Detect load URLs with `http` / `https` schemes.
- Automatically enable staging for such URLs.  

The right place to do both is probably in [LoadedTables.linkTable](https://github.com/UBOdin/mimir/blob/master/src/main/scala/mimir/data/LoadedTables.scala#L165).
